lib/Transforms/IPO/StructFieldCacheAnalysis.cpp

```diff
@@ -238,6 +254,44 @@ void StructFieldAccessManager::buildCloseProximityRelations() {
   }
 }
 
+void StructFieldAccessManager::suggestFieldReordering(bool UseOld) {
+  outs() << "------------- Suggestions on Reordering ------------------\n";
+  for (auto &it : CloseProximityBuilderMap) {
+    auto *type = it.first;
+    if (type->isLiteral()) {
+      outs() << "Recommendation on an anonymous struct ";
+    } else {
+      outs() << "Recommendation on struct [" << type->getStructName() << "] ";
+    }
+    auto Hotness = HotnessAnalyzer->getHotness(type);
+    assert(Hotness);
+    outs() << " (Hotness "
+           << 100 * Hotness.getValue() / HotnessAnalyzer->getMaxHotness()
+           << "% of hottest struct):\n";
+    auto *FRA =
+        UseOld
+            ? new OldFieldReorderAnalyzer(CurrentModule, type, it.second, NULL)
+            : new FieldReorderAnalyzer(CurrentModule, type, it.second, NULL);
+    FRA->makeSuggestions();
+  }
+  outs() << "----------------------------------------------------------\n";
+}
+
+void StructFieldAccessManager::suggestStructSplitting() {
+  outs() << "------------- Suggestions on Splitting ------------------\n";
+  for (auto &it : CloseProximityBuilderMap) {
+    auto *type = it.first;
+    if (type->isLiteral()) {
+      outs() << "Recommendation on an anonymous struct:\n";
+    } else {
+      outs() << "Recommendation on struct [" << type->getStructName() << "]:\n";
+    }
+    auto *SSA = new StructSplitAnalyzer(CurrentModule, type, it.second, NULL);
```

Comment:
I should print same thing for splitting suggestions. The reason I am adding hotness and other stuff is that I am using reordering in Chrome but I didn't use splitting so that's why I didn't pay attention to. I'll fix this.